server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -78,19 +82,71 @@ public void executeOrFail(DependencyCarrier dependencies,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) throws Exception {
-        BoundAddColumn stmt = bind(
-            alterTable,
-            plannerContext.transactionContext(),
-            dependencies.nodeContext(),
-            params,
-            subQueryResults,
-            dependencies.fulltextAnalyzerResolver());
+        if (plannerContext.clusterState().getNodes().getMinNodeVersion().onOrAfter(Version.V_5_1_0)) {
+            var addColumnRequest = createRequest(alterTable);
+
+            dependencies.alterTableOperation().executeAlterTableAddColumn(addColumnRequest)
+                .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
+        } else {
+            BoundAddColumn stmt = bind(
+                alterTable,
+                plannerContext.transactionContext(),
+                dependencies.nodeContext(),
+                params,
+                subQueryResults,
+                dependencies.fulltextAnalyzerResolver());
 
-        dependencies.alterTableOperation().executeAlterTableAddColumn(stmt)
-            .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
+
+            dependencies.alterTableOperation().executeAlterTableAddColumn(stmt)
+                .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
+        }
     }
 
     @VisibleForTesting
+    public static AddColumnRequest createRequest(AnalyzedAlterTableAddColumn alterTable) {
+        assert alterTable.analyzedTableElementsWithExpressions().columns().size() == 1 : "Add column must contain exactly one column";
+        AnalyzedColumnDefinition colToAdd = alterTable.analyzedTableElementsWithExpressions().columns().get(0);
+
+        // Reference creation is similar to AnalyzedTableElements.buildReference()
+        // but cannot use it as it passes fixed values for some params: ColumnPolicy.DYNAMIC, IndexType.PLAIN, nullable:true, hasDocValues:false
+        // which cannot be constant in case of add column.
+
+        DataType<?> type = colToAdd.dataType() == null ? DataTypes.UNDEFINED : colToAdd.dataType();
+        DataType<?> realType = ArrayType.NAME.equals(colToAdd.collectionType())
+            ? new ArrayType<>(type)
+            : type;
+
+        Reference ref = new SimpleReference(
+            new ReferenceIdent(alterTable.tableInfo().ident(), colToAdd.ident()),
+            RowGranularity.DOC, // column in ADD COLUMN cannot be PARTITIONED BY
+            realType,
+            colToAdd.objectType(), // Relevant only if added column has object type
```

Comment:
I initially had smth like `type.ID != Object.ID : strict :  colToAdd.objectType()` but I think this field is relevant only when adding object columns